examples/database-email-update/index.js

```diff
@@ -7,119 +7,134 @@
 
-// A JSON Object to hold all tasks in the Notion database.
-let tasksInDatabase = {}
+/**
+ * Local map to store task pageId to its last status.
+ * { [pageId: string]: string }
+ */
+const taskPageIdToStatusMap = {}
 
-async function findChangesAndSendEmails() {
-  console.log("Looking for changes in Notion database ")
+/**
+ * Initialize local data store.
+ * Then poll for changes every 5 seconds (5000 milliseconds).
+ */
+setInitialTaskPageIdToStatusMap().then(() => {
+  setInterval(findAndSendEmailsForUpdatedTasks, 5000)
+})
+
+/**
+ * Get and set the initial data store with tasks currently in the database.
+ */
+async function setInitialTaskPageIdToStatusMap() {
+  const currentTasks = await getTasksFromNotionDatabase()
+  for (const { pageId, status } of currentTasks) {
+    taskPageIdToStatusMap[pageId] = status
+  }
+}
+
+async function findAndSendEmailsForUpdatedTasks() {
   // Get the tasks currently in the database.
-  const currTasksInDatabase = await getTasksFromDatabase()
+  const currentTasks = await getTasksFromNotionDatabase()
+  // Return any tasks that have had their status updated.
+  const updatedTasks = findUpdatedTasks(currentTasks)
+  console.log(`Found ${updatedTasks.length} updated tasks.`)
+  // For each updated task, update local db and send an email notification.
+  for (const task of updatedTasks) {
+    taskPageIdToStatusMap[task.pageId] = task.status
+    await sendUpdateEmailWithSendgrid(task)
+  }
+}
 
-  // Iterate over the current tasks and compare them to tasks in our local store (tasksInDatabase).
-  for (const [key, value] of Object.entries(currTasksInDatabase)) {
-    const page_id = key
-    const curr_status = value.Status
-    // If this task hasn't been seen before.
-    if (!page_id in tasksInDatabase) {
-      // Add this task to the local store of all tasks.
-      tasksInDatabase[page_id] = {
-        Status: curr_status,
-      }
-    } else {
-      // If the current status is different from the status in the local store.
-      if (curr_status !== tasksInDatabase[page_id].Status) {
-        // Change the local store.
-        tasksInDatabase[page_id] = {
-          Status: curr_status,
-        }
-        // Send an email about this change..
-        const msg = {
-          to: process.env.EMAIL_TO_FIELD,
-          from: process.env.EMAIL_FROM_FIELD,
-          subject: "Notion Task Status Updated",
-          text:
-            "A Notion task's: " +
-            value.Title +
-            " status has been updated to " +
-            curr_status +
-            ".",
-        }
-        sendgridMail
-          .send(msg)
-          .then(() => {
-            console.log("Email Sent")
-          })
-          .catch(error => {
-            console.error(error)
-          })
-        console.log("Status Changed")
-      }
+/**
+ * Gets tasks from the database.
+ *
+ * Returns array of task objects with pageId, status, and title.
+ * Array<{ pageId: string, status: string, title: string }>
```

Comment:
jsdoc format?

Reply:
Is this not jsdoc? Is there a better way to show the return type?